src/WebJobs.Extensions.DurableTask/DefaultPlatformInformationProvider.cs

```diff
@@ -49,6 +44,60 @@ public bool InLinuxAppService()
             return inLinuxDedicated;
         }
 
+        public OperatingSystem GetOperatingSystem()
+        {
+            if (this.InLinuxConsumption() || this.InLinuxAppService())
+            {
+                return OperatingSystem.Linux;
+            }
+
+            return OperatingSystem.Windows;
+        }
+
+        public AppServicePlan GetAppSevicePlan()
+        {
+            if (this.InLinuxConsumption() | this.InWindowsConsumption())
```

Comment:
Seems like `||` (boolean operator with shortcircuiting) would be more appropriate than `|` here.

```suggestion
            if (this.InLinuxConsumption() || this.InWindowsConsumption())
```